src/lib/statistics/chunk_statistics/histograms/equal_width_histogram.cpp

```diff
@@ -270,7 +270,13 @@ BinID EqualWidthHistogram<T>::_bin_for_value(const T& value) const {
     if (_bin_data.bin_count_with_larger_range == 0u ||
         value <= _bin_maximum(_bin_data.bin_count_with_larger_range - 1u)) {
       // All bins up to that point have the exact same width, so we can use index 0.
-      return (value - _bin_data.minimum) / _bin_width(0u);
+      const auto bin_id = (value - _bin_data.minimum) / _bin_width(0u);
+
+      // The above calculation can lead to an index that is equal to or larger than the number of bins there are,
+      // due to floating point arithmetic.
+      // We checked before that the value is not larger than the maximum of the histogram,
+      // so simply return the last bin.
```

Comment:
"... so in that case, simply return the last bin"?

Reply:
Done.